app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt

```diff
@@ -0,0 +1,91 @@
+package com.twain.interprep.presentation.ui.components
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.twain.interprep.R
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TextFormInput(
+    modifier: Modifier,
+    value: String,
+    labelText: String,
+    bottomText: String = "",
+    required: Boolean = false,
+    errorMessage: String = "",
+) {
+    val label = if (required) "$labelText *" else labelText
+    var text by remember { mutableStateOf(value) }
+    var isError by remember { mutableStateOf(false) }
+
+    fun validate(text: String) {
+        isError = (required && text == "")
+    }
```

Comment:
I'm not sure if these should go in the component or if this logic will be in the screen. Please advise what the best way should be

Reply:
@sarahhjchung move validate function to utils class